MetaMorpheus/GUI/Util/Converters/AveragingTaskConverters.cs

```diff
@@ -0,0 +1,253 @@
+        }
+    }
+
+    public class SpectraFileAveragingTypeToNumberToAverageReadOnlyConverter : BaseValueConverter<SpectraFileAveragingTypeToNumberToAverageReadOnlyConverter>
+    {
+        public override object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            if (value.GetType() == typeof(SpectraFileAveragingType))
+            {
+                switch (value)
+                {
+                    case SpectraFileAveragingType.AverageAll:
+                        return true;
+
+                    case SpectraFileAveragingType.AverageEverynScans:
+                        return false;
+
+                    case SpectraFileAveragingType.AverageEverynScansWithOverlap:
+                        return false;
+
+                    case SpectraFileAveragingType.AverageDdaScans:
+                        return false;
+
+                    case SpectraFileAveragingType.AverageDdaScansWithOverlap:
+                        return false;
+                }
+            }
+
+            return false;
+        }
+
+        public override object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            throw new NotImplementedException();
+        }
+    }
+
+    public class SpectraFileAveragingTypeToNumberOpacityConverter : BaseValueConverter<SpectraFileAveragingTypeToNumberOpacityConverter>
+    {
+        private double doNotDisplay = 0.5;
+        public override object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            if (value.GetType() == typeof(SpectraFileAveragingType))
+            {
+                switch (value)
+                {
+                    case SpectraFileAveragingType.AverageAll:
+                        return doNotDisplay;
+
+                    case SpectraFileAveragingType.AverageEverynScans:
+                        return 1;
+
+                    case SpectraFileAveragingType.AverageEverynScansWithOverlap:
+                        return 1;
+
+                    case SpectraFileAveragingType.AverageDdaScans:
+                        return 1;
+
+                    case SpectraFileAveragingType.AverageDdaScansWithOverlap:
+                        return 1;
+                }
+            }
+
+            return false;
+        }
+
+        public override object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            throw new NotImplementedException();
+        }
+    }
+
+    public class OutlierRejectionTypeToSigmaReadOnlyConverter : BaseValueConverter<OutlierRejectionTypeToSigmaReadOnlyConverter>
+    {
+        public override object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            if (value.GetType() == typeof(OutlierRejectionType))
+            {
+                OutlierRejectionType val = (OutlierRejectionType)value;
```

Comment:
Maybe do a switch statement here, like you did above?

Reply:
Even with the newer C# versions ostensibly allowing you to do it, it's still kind of a pain to implement